### PR TITLE
CHOWTapeModel: init at unstable-2020-12-12

### PR DESCRIPTION
**Pull request by magnetophon**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    repo = "AnalogTapeModel";
+    rev = "a7cf10c3f790d306ce5743bb731e4bc2c1230d70";
+    sha256 = "09nq8x2dwabncbp039dqm1brzcz55zg9kpxd4p5348xlaz5m4661";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    pkgconfig
+  ];
+  buildInputs = [
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:32]:
```suggestion
    pkg-config
  ];

  buildInputs = [
```

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+, fetchFromGitHub
+, freeglut
+, freetype
+, libGL
+, libXcursor
+, libXext
+, libXinerama
+, libXrandr
+, libjack2
+, pkgconfig
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:12]:
```suggestion
, pkg-config
```

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  buildPhase = ''
+    cd Plugin/
+    ./build_linux.sh
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib/lv2 $out/lib/vst3 $out/bin $out/share/doc/${pname}/
+    cd Builds/LinuxMakefile/build/
+    cp ${pname}.a  $out/lib
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:54]: Please do not use pname in here. It makes grasping what is happening a bit hard.

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    cp -r ${pname}.lv2 $out/lib/lv2
+    cp -r ${pname}.vst3 $out/lib/vst3
+    cp ${pname}  $out/bin
+    cp ../../../../Manual/ChowTapeManual.pdf $out/share/doc/${pname}/
+  '';
+
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/jatinchowdhury18/AnalogTapeModel";
+    description = "Physical modelling signal processing for analog tape recording. LV2, VST3 and standalone";
+    license = with licenses; [ gpl3 ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:64]: Plus or Only?

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    mkdir -p $out/lib/lv2 $out/lib/vst3 $out/bin $out/share/doc/${pname}/
+    cd Builds/LinuxMakefile/build/
+    cp ${pname}.a  $out/lib
+    cp -r ${pname}.lv2 $out/lib/lv2
+    cp -r ${pname}.vst3 $out/lib/vst3
+    cp ${pname}  $out/bin
+    cp ../../../../Manual/ChowTapeManual.pdf $out/share/doc/${pname}/
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:61]:
```suggestion
  meta = with lib; {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/CHOWTapeModel/default.nix (hunk 4 above)
Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix:66]: Thanks for your tireless work @SuperSandro2000!

The other suggestions are now done.
I'll push a fix for this one when upstream responds.

Related, do you have an idea what is going on in [this issue](https://github.com/jatinchowdhury18/AnalogTapeModel/issues/164)?

Comment (SuperSandro2000) [reply pkgs/applications/audio/CHOWTapeModel/default.nix:66]: It's probably gpl3Only.

Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix]: OK, I pushed that.

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -20636,6 +20636,8 @@ in
 
   bookworm = callPackage ../applications/office/bookworm { };
 
+  CHOWTapeModel = callPackage ../applications/audio/CHOWTapeModel {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:20639]:
```suggestion
  CHOWTapeModel = callPackage ../applications/audio/CHOWTapeModel { };
```

File: pkgs/applications/audio/CHOWTapeModel/default.nix [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    cp CHOWTapeModel  $out/bin
+    cp ../../../../Manual/ChowTapeManual.pdf $out/share/doc/CHOWTapeModel/
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/jatinchowdhury18/AnalogTapeModel";
+    description = "Physical modelling signal processing for analog tape recording. LV2, VST3 and standalone";
+    license = with licenses; [ gpl3Only ];
+    maintainers = with maintainers; [ magnetophon ];
+    platforms = with platforms; linux;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/CHOWTapeModel/default.nix:68]:
```suggestion
    platforms = platforms.linux;
```

File: pkgs/applications/audio/CHOWTapeModel/default.nix (hunk 4 above)
Comment (magnetophon) [reply pkgs/applications/audio/CHOWTapeModel/default.nix]: Turns out we guessed wrong...

**Pull request merged**